File: auth-server/src/main/java/no/vicx/authserver/config/DefaultUserProperties.java
package no.vicx.authserver.config;

import jakarta.validation.constraints.NotBlank;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@Validated
@ConfigurationProperties("default-user")
public record DefaultUserProperties(
        @NotBlank String username,
        @NotBlank String password,
        @NotBlank String name,
        @NotBlank String email) {
}

File: auth-server/src/main/kotlin/no/vicx/authserver/config/DefaultUserProperties.kt
package no.vicx.authserver.config

import jakarta.validation.constraints.NotBlank
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties("default-user")
class DefaultUserProperties {
    @NotBlank
    lateinit var username: String

    @NotBlank
    lateinit var password: String

    @NotBlank
    lateinit var name: String

    @NotBlank
    lateinit var email: String
}
